ext/libdatadog_api/feature_flags.c

```diff
@@ -0,0 +1,430 @@
+  
+  if (!resolution_details) {
+    return Qnil;
+  }
+
+  // Use the new FFI function to get the value
+  struct ddog_ffe_VariantValue value = ddog_ffe_assignment_get_value(*resolution_details);
+  
+  switch (value.tag) {
+    case DDOG_FFE_VARIANT_VALUE_NONE:
+      return Qnil;
+    case DDOG_FFE_VARIANT_VALUE_STRING:
+      return rb_str_new((const char*)value.string.ptr, value.string.len);
+    case DDOG_FFE_VARIANT_VALUE_INTEGER:
+      return LONG2NUM(value.integer);
+    case DDOG_FFE_VARIANT_VALUE_FLOAT:
+      return rb_float_new(value.float_);
+    case DDOG_FFE_VARIANT_VALUE_BOOLEAN:
+      return value.boolean ? Qtrue : Qfalse;
+    case DDOG_FFE_VARIANT_VALUE_OBJECT:
+      return rb_str_new((const char*)value.object.ptr, value.object.len);
+    default:
+      return Qnil;
+  }
+}
+
+static VALUE resolution_details_get_reason(VALUE self) {
+  ddog_ffe_Handle_ResolutionDetails *resolution_details;
+  TypedData_Get_Struct(self, ddog_ffe_Handle_ResolutionDetails, &resolution_details_typed_data, resolution_details);
+  
+  if (!resolution_details) {
+    return Qnil;
+  }
+
+  // Use the new FFI function to get the reason
+  enum ddog_ffe_Reason reason = ddog_ffe_assignment_get_reason(*resolution_details);
+  
+  switch (reason) {
+    case DDOG_FFE_REASON_STATIC:
+      return ID2SYM(rb_intern("static"));
+    case DDOG_FFE_REASON_DEFAULT:
+      return ID2SYM(rb_intern("default"));
+    case DDOG_FFE_REASON_TARGETING_MATCH:
+      return ID2SYM(rb_intern("targeting_match"));
+    case DDOG_FFE_REASON_SPLIT:
+      return ID2SYM(rb_intern("split"));
+    case DDOG_FFE_REASON_DISABLED:
+      return ID2SYM(rb_intern("disabled"));
+    case DDOG_FFE_REASON_ERROR:
+      return ID2SYM(rb_intern("error"));
+    default:
+      return Qnil;
+  }
+}
+
+static VALUE resolution_details_get_error_code(VALUE self) {
+  ddog_ffe_Handle_ResolutionDetails *resolution_details;
+  TypedData_Get_Struct(self, ddog_ffe_Handle_ResolutionDetails, &resolution_details_typed_data, resolution_details);
+  
+  if (!resolution_details) {
+    return Qnil;
+  }
+
+  // Use the new FFI function to get the error code
+  enum ddog_ffe_ErrorCode error_code = ddog_ffe_assignment_get_error_code(*resolution_details);
+  
+  switch (error_code) {
+    case DDOG_FFE_ERROR_CODE_TYPE_MISMATCH:
+      return ID2SYM(rb_intern("type_mismatch"));
+    case DDOG_FFE_ERROR_CODE_PARSE_ERROR:
+      return ID2SYM(rb_intern("parse_error"));
+    case DDOG_FFE_ERROR_CODE_FLAG_NOT_FOUND:
+      return ID2SYM(rb_intern("flag_not_found"));
+    case DDOG_FFE_ERROR_CODE_TARGETING_KEY_MISSING:
+      return ID2SYM(rb_intern("targeting_key_missing"));
+    case DDOG_FFE_ERROR_CODE_INVALID_CONTEXT:
+      return ID2SYM(rb_intern("invalid_context"));
+    case DDOG_FFE_ERROR_CODE_PROVIDER_NOT_READY:
+      return ID2SYM(rb_intern("provider_not_ready"));
+    case DDOG_FFE_ERROR_CODE_GENERAL:
+      return ID2SYM(rb_intern("general"));
+    default:
+      return Qnil;
+  }
```

Comment:
minor: "general" is a safer default
```suggestion
  switch (error_code) {
    case DDOG_FFE_ERROR_CODE_OK:
      return Qnil;
    case DDOG_FFE_ERROR_CODE_TYPE_MISMATCH:
      return ID2SYM(rb_intern("type_mismatch"));
    case DDOG_FFE_ERROR_CODE_PARSE_ERROR:
      return ID2SYM(rb_intern("parse_error"));
    case DDOG_FFE_ERROR_CODE_FLAG_NOT_FOUND:
      return ID2SYM(rb_intern("flag_not_found"));
    case DDOG_FFE_ERROR_CODE_TARGETING_KEY_MISSING:
      return ID2SYM(rb_intern("targeting_key_missing"));
    case DDOG_FFE_ERROR_CODE_INVALID_CONTEXT:
      return ID2SYM(rb_intern("invalid_context"));
    case DDOG_FFE_ERROR_CODE_PROVIDER_NOT_READY:
      return ID2SYM(rb_intern("provider_not_ready"));
    case DDOG_FFE_ERROR_CODE_GENERAL:
      return ID2SYM(rb_intern("general"));
    default:
      return ID2SYM(rb_intern("general"));
  }
```